### PR TITLE
Add markdown feature

### LINKED ISSUE #10: [Req]: Markdown Preview Tool
### Contact Details

_No response_

### Explain the proposed feature

Markdown preview tool

### Explain the possible use cases for the feature

_No response_

### Any other details you would like to provide?

_No response_

### PR DESCRIPTION
**Pull request by diego-rapoport**
## Type of Pull Request

- [ ] Bug fix
- [x] Enhancement

Related Issue #s or links (if any): 

## Description of Changes
Just added a Markdown Previewer to the menu. Fully functional. If more tests are needed, please let me know.
Closes #10

### CODE REVIEW

**Review by dlford — CHANGES_REQUESTED** [review]
@diego-rapoport Great job on this! I think we just need to have the layout stack on mobile, it gets too tight having it side by side.

**Comment by diego-rapoport** [comment]
> @diego-rapoport Great job on this! I think we just need to have the layout stack on mobile, it gets too tight having it side by side.

Thank you very much! I'm so glad I could contribute.
 I'll make those changes right away

**Comment by diego-rapoport** [comment]
@dlford How about that?

Recording 2022-10-08 at 22 37 42

I had to make a global css on this, but it's really simple and straight forward.

**Comment by dlford** [comment]
@diego-rapoport I think this would be cleaner without the CSS file, you can accomplish this with the `sx` prop on MUI components (https://mui.com/system/getting-started/the-sx-prop/).

Also, looks like there is a linting error in the latest deploy log.

**Comment by diego-rapoport** [comment]
Oh, sorry. I thought I had done the push to my fork only. My mistake.

**Comment by dlford** [comment]
> Oh, sorry. I thought I had done the push to my fork only. My mistake.

No worries, you did, Netlify deploys previews of PR code automatically.

**Review by dlford — CHANGES_REQUESTED** [review]
@diego-rapoport this looks good, can you please do the following so we can get it merged:
1. Run `npm install` (looks like `package-lock.json` is out of sync, this will fix it)
2. Run `npm run cq` and address any issues
3. Commit and push those changes

**Comment by dlford** [comment]
@diego-rapoport not sure if you saw my last comment? We just need to fix those couple things and then I can merge this PR.

**Comment by diego-rapoport** [comment]
> @diego-rapoport not sure if you saw my last comment? We just need to fix those couple things and then I can merge this PR.

Hey! I'm truly sorry. Just got really busy at work and still couldn't make it. I hope by the end of the week I could focus on that.

**Comment by dlford** [comment]
@diego-rapoport no worries, thanks!

**Comment by diego-rapoport** [comment]
I really thought that I had the Lint and Test phase right. If I run locally my fork(already did the pull and updated it) `npm ci` it doesn't give me any errors.
Can you help me find what's wrong @dlford? I already updated the `package-lock.json` file and it seamed to be good.

File: __TESTS__/markdown.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+<p>  This is just a paragraph<br>  with a small linebreak inside</p>
+<ul>
+<li>My first item</li>
+<li>Second item</li>
+</ul>
+`;
+
+test('previews markdown correctly', async () => {
+  render(<MarkDownPreview />);
+  const input = screen.getByRole('textbox');
```
Comment (dlford) [inline __TESTS__/markdown.spec.tsx:26]: This test is failing because there are two text boxes on the page, let's use `getByLabelText` here instead.

**Review by dlford — CHANGES_REQUESTED** [review]
@diego-rapoport Just one minor issue with the test, then we are good to merge, thanks!

**Comment by dlford** [comment]
@diego-rapoport sorry to nit-pick here, but can we also give that input a label for accessibility?

**Comment by diego-rapoport** [comment]
> @diego-rapoport sorry to nit-pick here, but can we also give that input a label for accessibility?

Sure! 
What label should I give? How should I place it as well?
If you can give me more details about how and what should I put that would be great.

File: pages/markdown.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        className='mainBox'
+        display='flex'
+        justifyContent='stretch'
+        gap={3}
+        width='80%'
+        sx={{
+          flexDirection: { md: 'row', xs: 'column' },
+        }}
+      >
+        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
```
Comment (dlford) [inline pages/markdown.tsx:32]: delete this line to re-enable the lint rule

File: pages/markdown.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        <label htmlFor='inputText' />
```
Comment (dlford) [inline pages/markdown.tsx:33]: Delete this empty label tag

File: pages/markdown.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        <TextField
+          id='inputText'
```
Comment (dlford) [inline pages/markdown.tsx:35]: Add a `label` attribute, like the examples here - https://mui.com/material-ui/react-text-field/

**Review by dlford — CHANGES_REQUESTED** [review]

File: pages/markdown.tsx (hunk 4 above)
Comment (dlford) [reply pages/markdown.tsx:34]: Let's make that a meaningful label, like "Markdown"?

**Review by dlford — CHANGES_REQUESTED** [review]

File: pages/markdown.tsx (hunk 4 above)
Comment (diego-rapoport) [reply pages/markdown.tsx:34]: That's why I asked **what** to write... but ok.

**Pull request merged**